jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -484,6 +488,19 @@ public JavaContainerBuilder setMainClass(String mainClass) {
     return this;
   }
 
+  /**
+   * Sets the modification time provider for container files.
+   *
+   * @param modificationTimeProvider a provider that takes a source path and destination path on the
+   *     container and returns the file modification time that should be set for that path
+   * @return this
+   */
+  public JavaContainerBuilder setModificationTimeProvider(
```

Comment:
I'm not sure how I feel about adding methods like these to `JavaContainerBuilder`. I feel like it's best to keep this class as simple as possible and delegate the non-java-specific stuff to `JibContainerBuilder` or something else, but I guess this is necessary since the timestamp provider needs to be applied as each file is added?

Reply:
Yeah, files are added recursively inside `JavaContainerBuilder`, so I also find passing a provider necessary in its current state.